src/coreclr/src/nativeaot/Runtime/amd64/StubDispatch.asm

```diff
@@ -85,6 +91,11 @@ LEAF_END RhpVTableOffsetDispatch, _TEXT
 ;; Initial dispatch on an interface when we don't have a cache yet.
 LEAF_ENTRY RhpInitialInterfaceDispatch, _TEXT
 ALTERNATE_ENTRY RhpInitialDynamicInterfaceDispatch
+        ;; Trigger an AV if we're dispatching on a null this.
+        ;; The exception handling infrastructure is aware of the fact that this is the first
+        ;; instruction of RhpInitialInterfaceDispatch and uses it to translate an AV here
+        ;; to a NullReferenceException at the callsite.
+        cmp     qword ptr [rcx], 0
```

Comment:
This is technically not necessary since the AV will happen in Runtime.Base, but it's going to throw the Runtime.Base's NullReferenceException that would not be catchable in the original Redhawk model.